hw/ip/otbn/dv/otbnsim/sim/state.py

```diff
@@ -257,13 +257,18 @@ def post_insn(self, loop_warps: Dict[int, int]) -> None:
         if self._err_bits:
             self.pending_halt = True
 
-        # Check the next PC is valid, but only if we're not stopping anyway. We
-        # do this right at the end of post_insn because any of the errors that
-        # we've just checked for would squash the "next PC" check.
-        if not self.pending_halt:
-            if not self.is_next_pc_valid():
-                self._err_bits |= ErrBits.BAD_INSN_ADDR
-                self.pending_halt = True
+        # Check the next PC is valid, but only if we're not stopping anyway.
```

Comment:
```suggestion
        # Check if the next PC is valid, but only if we're not stopping anyway.
```

Reply:
Hmm, I really mean something more like "assert that...", but maybe it's clearer if I change it to "Check that the next PC..."